### ISSUE TITLE
Carousel implementation does not pass WCAG2a using Axe tests in Cypress

### ISSUE THREAD

**Issue by kylane** [express-labs/pure-react-carousel#318]
- `pure-react-carousel` version: ^1.27.5
- `react` version: ^16.8
- `browser` used: Chrome
- `node` version: v12.18.4

### Relevant code or config:

```js
            <div
                data-testid="spotlights"
                style={{ height: '100%', position: 'relative' }}
                role="region"
                aria-label="Spotlights carousel"
            >
                <CarouselProvider
                    visibleSlides={1}
                    totalSlides={totalSlides}
                    step={1}
                    naturalSlideWidth={1967}
                    naturalSlideHeight={721}
                    isPlaying={!account}
                    interval={10000}
                    style={{ height: '100%' }}
                >
                    {totalSlides > 1 && (
                        <div
                            style={{
                                width: 36,
                                height: '100%',
                                position: 'absolute',
                                top: 4,
                                left: 4,
                                zIndex: 101,
                            }}
                        >
                            <ButtonBack>
                                <ChevronLeftIcon />
                            </ButtonBack>
                        </div>
                    )}
                    {totalSlides > 1 && (
                        <div
                            style={{
                                width: 36,
                                height: '100%',
                                position: 'absolute',
                                top: 4,
                                right: 0,
                                zIndex: 101,
                            }}
                        >
                            <ButtonNext>
                                <ChevronRightIcon />
                            </ButtonNext>
                        </div>
                    )}
                    {totalSlides > 1 && (
                        <div
                            style={{
                                width: 36,
                                position: 'absolute',
                                top: 4,
                                left: 32,
                                zIndex: 101,
                            }}
                        >
                            <ButtonPlay childrenPlaying={<PauseIcon />} childrenPaused={<PlayArrowIcon />} />
                        </div>
                    )}
                    {totalSlides > 1 && (
                        <div
                            style={{
                                width: 200,
                                marginLeft: -100,
                                position: 'absolute',
                                top: 0,
                                left: '50%',
                                zIndex: 101,
                                textAlign: 'center',
                            }}
                        >
                            <DotGroup />
                        </div>
                    )}
                    <div
                        style={{
                            zIndex: 99,
                        }}
                    >
                        <Slider>
                            {slides.map((item, index) => {
                                return (
                                    <Slide index={item.index} key={index}>
                                        <a href={item.link} aria-label={item.aria}>
                                            <Image
                                                src={item.src}
                                                alt={item.alt}
                                                style={{ width: '100%' }}
                                                aria-label={item.aria}
                                            />
                                        </a>
                                    </Slide>
                                );
                            })}
                        </Slider>
                    </div>
                </CarouselProvider>
            </div>
```

### What you did:

Running Axe in Cypress tests for WCAG compliance.

### What happened:

```
failureSummary: "Fix any of the following:↵  aria-label attribute does not exist or is empty↵  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty↵  Element has no title attribute or the title attribute is empty"
html: "<div class="horizontalSlider___281Ls carousel__slider carousel__slider--horizontal" aria-live="polite" tabindex="0" role="listbox">"
impact: "serious"
```
```
failureSummary: "Fix all of the following:↵  List element has direct children that are not allowed inside <li> elements"
html: "<ul class="sliderTray___-vHFQ sliderAnimation___300FY carousel__slider-tray carousel__slider-tray--horizontal" style="width: 400%; transform: translateX(0%) translateX(0px); flex-direction: row;">"
impact: "serious"
```
### Problem description:

Does not pass WCAG2a tests at the moment.

### Suggested solution:

Fixes are explained in the errors above.

### Can you help us fix this issue by submitting a pull request?

TBH - Have never tried - but happy to when I get some time.

**Comment by kylane** [comment]
Sorry - inadvertant clicking....

**Comment by CatchGus** [comment]
The first violation can be solved by passing your own `aria-label` prop to `Slider`
```
<Slider aria-label="My accessible name">
```

The second violation is "List element has direct children with a role that is not allowed: option". Apparently putting a role of `option` on an `li` element is a no no. Yet to find a workaround for this.

**Comment by kylane** [comment]
> Should probably be fixed up the code though... Also this is not a Cypress only issue, I also saw these violations surfaced in our Storybook.

Appreciate the help - I was using trayProps to try set the aria labels - my bad.

**Comment by tim-steele** [comment]
@CatchGus Just so I understand, the issue is because we are defaulting to <li> and <li> is not allowed the role `option`?

**Comment by CatchGus** [comment]
> @CatchGus Just so I understand, the issue is because we are defaulting to `<li>` and `<li>` is not allowed the role `option`?

@tim-steele Yes, that seems to be the case. I guess it is because it has conflicting roles of `listitem` and `option`.

**Comment by CatchGus** [comment]
You can see more details of the violations if you install the [Axe Chrome extension](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd) and run it over your [examples page](https://express-labs.github.io/pure-react-carousel/).

**Comment by tim-steele** [comment]
@CatchGus I will mark it as an issue - would you be willing to do a pull request to make it a `<div>` or an item that will pass the WCAG markup?

I am wondering if this would be a breaking change. If anyone is using `li` CSS selectors, it would break if they are expecting the default `li`.
